src/ServicePulse.Host/vue/src/components/monitoring/EndpointListSortableColumn.vue

```diff
@@ -1,19 +1,81 @@
 <script setup>
-defineProps({
+import { ref, computed } from "vue";
+import { useMonitoringStore } from "../../stores/MonitoringStore";
+
+const props = defineProps({
   unit: String,
+  isActive: Boolean,
+  sortBy: {
+    type: String,
+    required: true,
+    validator: (value) => ["name", "queueLength", "throughput", "retries", "processingTime", "criticalTime"].includes(value),
+  },
 });
+
+const monitoringStore = useMonitoringStore();
+const isActive = computed(() => props.isActive);
```

Comment:
You might consider passing in the active header as a `v-model` and then computing `isActive` off of that. If you do that then:
- You don't need a click handler on the parent
- The parent doesn't need to decide if this is active, the component itself can
- The parent can put a watch on the active column and update the monitoring store

This would make this a generic component that isn't unique to the EndpointList.
You _would_ need to update `activeColumn` in this component and you would need to expose the sort direction somehow. 

In this component:
```js

const activeColumn = defineModel({
  type: String,
  required: true,
});

const isActive = computed(() => activeColumn.value == props.sortBy);

// ...

function toggleSort() {
  activeColumn.value = props.sortBy;
  isAscending.value = isActive.value ? !isAscending.value : false;

  // ...
}
```

In the consumer component:
```js
<EndpointListSortableColumn 
  :sort-by="'queueLength'"
  v-model="activeColumn"
  v-tooltip
  title="Queue length: The number of messages waiting to be processed in the input queue(s) of the endpoint.">
  Queue Length
  <template #unit>(MSGS)</template>
</EndpointListSortableColumn
```

Reply:
This really helped me to look at this differently.  Thanks.  I was able to refactor a lot of stuff after reviewing this.